### PR TITLE
FXIOS-1456 fixes #7933 adds Save to Files to share sheet for PDF

### LINKED ISSUE #7933: FXIOS-1456 ⁃ Files cannot be shared from Browser on iOS
There's a big issue with the browser not being able to send files from the browser itself.
When trying to share a downloaded file from the Firefox browser app it will only send a path from the file system instead of the file itself.
For example when trying to share a downloaded PDF it will send something like that:

file:///private/var/mobile/Containers/Data/Application/A722-4516-AB70-4CDBD36FDB7A/tmp/TempDocs/PDF%20Project.pdf

The only way to share a file is to go from the browser to the Firefox download folder in the files app and share it from there.
It's very annoying because people who love the browser cannot be as productive with it as they would be with safari browser for example.

Please fix this so we can use it as we should.

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/FXIOS-1456)

### PR DESCRIPTION
**Pull request by ageofadz**
Fixes #7933 by adding Save to Files to share sheet if a PDF is shared. This is done by passing the URL directly in as activityItems in the activityViewController in ShareExtensionHelper.

IMG_1933

### CODE REVIEW

**Comment by bentsionmp** [comment]
Any idea if it'll share all types of files? Like .Doc or just PDF when loaded?
Also when what version is expected to fix this in public release?

**Comment by wachabaha** [comment]
[**_****_**](https://apps.apple.com/de/app/butter-royale/id1464623715)

**Review by wachabaha — APPROVED** [review]

**Comment by wachabaha** [comment]
https://apps.apple.com/de/app/butter-royale/id1464623715

**Comment by ageofadz** [comment]
> Any idea if it'll share all types of files Like .Doc or just PDF when loaded?
> Also when what version is expected to fix this in public release?

These changes work with .doc files - as for release schedule, I believe that is up to Mozilla

File: Client/Frontend/Share/ShareExtensionHelper.swift
Comment (electricRGB) [inline Client/Frontend/Share/ShareExtensionHelper.swift:50]:
```suggestion
        var activityViewController: UIActivityViewController
        if isFile(url: url) {
            activityViewController = UIActivityViewController(activityItems: [url], applicationActivities: nil)
        } else {
            activityViewController = UIActivityViewController(activityItems: activityItems, applicationActivities: nil)
        }
```

**Review by electricRGB — CHANGES_REQUESTED** [review]

**Review by electricRGB — APPROVED** [review]

**Comment by isabelrios** [comment]
Please do not merget yet...Looking at this PR since build failed to build and run the tests, see logs in here: 
https://app.bitrise.io/build/5f973a7aab57056c#?tab=log

**Comment by electricRGB** [comment]
@isabelrios I rebased their branch and rerunning BR to see if it works..

**Comment by electricRGB** [comment]
@isabelrios BR failed again, but it's failing on `testPromptPassCodeUponReentry` which is not touched by this PR at all. Will wait for your approval before merging, but I think it's safe.

**Comment by isabelrios** [comment]
> @isabelrios BR failed again, but it's failing on `testPromptPassCodeUponReentry` which is not touched by this PR at all. Will wait for your approval before merging, but I think it's safe.

yes, it is not related at all to this changes but I have seen it failing 3 out of 3 times, let me investigate it before merging please. If these changes are important to land, then please disable the tests. I will try to come back to this later today

**Comment by electricRGB** [comment]
Yep. Will wait. These can go in at any point this week or so. :)

**Comment by isabelrios** [comment]
I tried this locally and test is working, not sure what's happening on BR for that test and this PR, it is green for the rest. Feel free to land this and I will keep and eye on that. Thanks and sorry for the noise here :)

**Comment by electricRGB** [comment]
No worries. Thanks so much Isabel! :)

**Pull request merged**
